webapp-django/crashstats/crashstats/middleware.py

```diff
@@ -35,18 +35,21 @@ def process_request(self, request):
 
 
 class Pretty400Errors:
+    """Convert HTTP 400 errors to JSON if client doesn't accept html"""
+
     def __init__(self, get_response):
         self.get_response = get_response
 
     def __call__(self, request):
         response = self.get_response(request)
 
+        is_ajax = request.headers.get("x-requested-with") == "XMLHttpRequest"
         if (
             response.status_code == 400
-            and not request.is_ajax()
```

Comment:
They removed `is_ajax`. I think it was JS framework specific. The method didn't do much, so I reproduced it above. That maintains our existing behavior. Going forward, if we switch JS frameworks, we might need to fix this.